python-backend/app/api/parties/party/resources/party_resource.py

```diff
@@ -171,11 +114,26 @@ def put(self, party_guid):
         for key, value in data.items():
             setattr(existing_party, key, value)
 
+        # This condition should never pass, as every party should be
+        # associated with an address, but it's been added for runtime safety,
+        # in case the API is circumvented and a party is created without an
+        # address
+        if len(existing_party.address) == 0:
+            address = Address.create()
+            existing_party.address.append(address)
+
+        for key, value in data.items():
+            setattr(existing_party.address[0], key, value)
```

Comment:
should this exclude party_type_code? I don't think we want to all the API to change an Org to a Person or visa versa? Always check the parser arguments to see if there is a value we need to protect.

Reply:
Valid point

Reply:
This loop is duplicated above, but you have added the protection to type code. thanks!

Reply:
One look is for the `Party` instance. This one is for the associated `PartyAddress` instance. I've added the protection to the one above because it lives on `Party`.

Reply:
address has an address_id field, we don't serve it, but someone writing manual requests might be able to cause a real problem in editing the PK. we should protect against that. you are right though, i miss read the issue two loops.

Reply:
Maybe I'm wrong, but the parser doesn't know about `address_id`, so we don't protect against it.

Reply:
Wow, didn't have enough coffee today apparently, yup you are right. thanks.